### ISSUE TITLE
When using `bun install --yarn`, bun.lockb and yarn.lock start referring to public registry URLs instead of the configured private registry URLs

### ISSUE THREAD

**Issue by soneymathew** [oven-sh/bun#5370]
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Darwin 22.4.0 arm64 arm

### What steps can reproduce the bug?

1. Configure bunfig.toml to use private registries following https://bun.sh/docs/runtime/configuration#private-scopes-and-registries, ensure than no general public registries are configured in this.
2. Run `bun install --yarn` in a project with `yarn.lock` file having package references to these private registries

### What is the expected behavior?

bun will retain URL references to private registries

### What do you see instead?

Observe than bun is replacing private registry URLs with public registry URLs in `yarn.lock` file content

**Comment by taranek** [comment]
+1 from my side, experiencing the same issue now. 
The config I use:
```
[install]
registry = { url = "https://nexus.MY-ORGANISATION.net/repository/npm-registry/", token="$TOKEN" }
[install.lockfile]
print = "yarn"
```
running `bun install` overrides `yarn.lock` file with 175 entries (out of 912 total) pointing to `https://registry.npmjs.org` registry. 

The problem does not occur when running `yarn install` (912/912 entries point to the private registry).

Tried removing local and global cache, but no luck so far - the issue persists

**Comment by taranek** [comment]
@Electroid Is there a plan to fix it soon? That's the only blocker I got to use Bun in production.

**Comment by anru** [comment]
I'd say this bug is a blocker for usage bun as package install manager for many companies with private registries

**Comment by dylan-conway** [comment]
Fixed in #11606 and will be included in v1.1.13. Note that `bun.lockb` will need to be deleted and regenerated. If you continue have problems please reopen this issue or open a new one.
